File: technical/leet/findPeakElement.cpp
// 162
#include <vector>;
using namespace std;

class Solution {
public:
    int findPeakElement(vector<int>& nums) {
        int idx = 0;
        if(nums.size() < 2 || nums[0] > nums[1]) {
            return(0);
        }
        if(nums[nums.size() - 1] > nums[nums.size() - 2]) {
            return(nums.size() - 1);
        }
        while(true) {
            if(nums[idx] > nums[idx + 1] && nums[idx] > nums[idx - 1]) {
                return(idx);
            }
            idx++;
        }
    }
};

File: lsprangers-blog/docs/technical writing/leet/findPeakElement.py
#  162
class Solution:
    def findPeakElement(self, nums: List[int]) -> int:
        if len(nums) < 2:
            return(0)
            
        for idx, num in enumerate(nums):
            if idx == 0:
                if num > nums[idx + 1]:
                    return(idx)
            elif idx == len(nums) - 1:
                if num > nums[idx - 1]:
                    return(idx)
            
            else:
                if num > nums[idx - 1] and num > nums[idx + 1]:
                    return(idx)
        
        return(-1)